torch/_refs/linalg/__init__.py

```diff
@@ -97,6 +97,34 @@ def diagonal(
     return torch.diagonal(input, offset=offset, dim1=dim1, dim2=dim2)
 
 
+def _check_vector_norm_args(
+    x: TensorLikeType, ord: Union[float, int] = 2, dim: Optional[DimsType] = None
+):
+    from torch.fx.experimental.symbolic_shapes import sym_or
+
+    if not (ord < 0.0 or ord == float("inf")):
+        return
+
+    torch._check(
+        sym_or(
+            x.numel() != 0,
+            not isinstance(dim, IntLike) and dim is not None and len(dim) != 0,
+        ),
+        "linalg.vector_norm cannot compute the {ord} norm on an empty tensor "
```

Comment:
```suggestion
        lambda: (f"linalg.vector_norm cannot compute the {ord} norm on an empty tensor "
```

Reply:
this is intentional torch.compile does not support lambda strings here.
https://github.com/pytorch/pytorch/issues/149997